pulp_smash/tests/rpm/cli/test_upload.py

```diff
@@ -14,7 +21,115 @@ def setUpModule():  # pylint:disable=invalid-name
     utils.pulp_admin_login(config.get_config())
 
 
-class UploadDrpmTestCase(unittest.TestCase):
+class _BaseCLIDownloadTestCase(unittest.TestCase):
+
+    @classmethod
+    def setUpClass(cls):
+        cls.cfg = config.get_config()
+        cls.client = cli.Client(cls.cfg)
+
+    def temp_download(self, file_url, filename='tmp.rpm'):
+        """Download a file from the given url to a temp directory.
+
+        Args:
+            file_url: String. The URL of the file to download. Required
+            filename: String. Assigns a name to the file downloaded. Optional.
+
+        Returns:
+            String. Full path to file downloaded.
+
+        After the test case runs, the temp directory and file is destroyed.
+
+        """
+        temp_dir = self.client.run('mktemp --directory'.split()).stdout.strip()
+        self.addCleanup(self.client.run, 'rm -rf {}'.format(temp_dir).split())
+        file_path = os.path.join(temp_dir, os.path.split(filename)[-1])
+        try:
+            self.client.run(
+                'curl -o {} {}'.format(file_path, file_url).split()
+            )
+        except urllib.error.HTTPError:
+            raise self.skipTest('Skipping test, could not download resource')
+
+        return file_path
+
+    def create_repo(self, plugin, repo_id):
+        """Create a repo with given plugin.
+
+        Args:
+            plugin: String. Name of plugin. Required
+            repo_id: String. Name of repo to be created. Required
+
+        Returns:
+            Completed process
+
+        After the test case runs, the repo is destroyed.
+
+        """
+        result = self.client.run(
+            'pulp-admin {} repo create --repo-id {}'
+            .format(plugin, repo_id).split()
+        )
+        self.addCleanup(
+            self.client.run,
+            'pulp-admin {} repo delete --repo-id {}'
+            .format(plugin, repo_id).split()
+        )
+        return result
+
+    def upload_to_repo(self, **kwargs):
+        """Upload a file to a repo.
+
+        Args:
+            plugin: String. Name of plugin. Required
+            filetype: String. Name of filetype. Required
+            filename: String. Full path of file. Required
+            args: String. Additional arguments to uploads command. Optional
+
+        Returns:
+            Completed process.
+
+        """
+        if 'args' not in kwargs.keys():
+            kwargs['args'] = ''
+        return self.client.run(
+            'pulp-admin {} repo uploads {} --repo-id {} --file {} {}'
+            .format(
+                kwargs['plugin'],
+                kwargs['filetype'],
+                kwargs['repo_id'],
+                kwargs['filename'],
+                kwargs['args']
```

Comment:
This isn't a great way to use kwargs. What happens if `kwargs['filetype']` isn't present? A KeyError exception is thrown. Given that these arguments are required, it's best to just list them as required arguments.

```python
upload_to_repo(self, plugin, filetype, repo_id, ...)
```